test/ao/sparsity/test_structured_sparsifier.py

```diff
@@ -639,3 +663,92 @@ def test_complex_conv2d(self):
                     torch.device(device),
                     also_prune_bias,
                 )
+
+    def test_prune_lstm_linear_multiple_layer(self):
+        """
+        Test fusion support for LSTM(multi-layer) -> Linear
+        """
+        model = LSTMLinearModel(
+            ntoken=10,
+            ninp=8,
+            nhid=8,
+            nlayers=2,
+        )
+
+        config = [
+            {"tensor_fqn": "rnn.weight_ih_l0"},
+            {"tensor_fqn": "rnn.weight_hh_l0"},
+            {"tensor_fqn": "rnn.bias_ih_l0"},
+            {"tensor_fqn": "rnn.bias_hh_l0"},
+            {"tensor_fqn": "rnn.weight_ih_l1"},
+            {"tensor_fqn": "rnn.weight_hh_l1"},
+            {"tensor_fqn": "rnn.bias_ih_l1"},
+            {"tensor_fqn": "rnn.bias_hh_l1"},
+        ]
+
+        rnn_input = torch.ones((1, 8))
+        fx_pruner = BottomHalfPruner({"sparsity_level": 0.5})
+        fx_pruner.prepare(model, config)
+
+        fx_pruner.enable_mask_update = True
+        fx_pruner.step()
+
+        model.eval()
+        _, _ = model(rnn_input)
+        pruned_model = fx_pruner.prune()
+        pruned_model.eval()
+        _, _ = pruned_model(rnn_input)
+
+        expected_params = dict(model.named_parameters())
+        for name, param in model.named_parameters():
```

Comment:
If anyone has a better idea about how to test this, please let me know.

Reply:
It appears you are testing the parameters against themselves there? Do you mean to iterate over the pruned_model?

Reply:
If so, I believe this is equivalent to your test:
```
pruned_params = dict(pruned_model.named_parameters())
for name, param in model.named_parameters():
    assert name in pruned_params, f"pruned param {name} is missing"
    assert elements_are_subset(pruned_params[name], param), f"pruned param {name} ({pruned_params[name].shape}) is not a subset of model ({param.shape})"
```

Reply:
thanks, good catch!